WPFCommandFactory/Command.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Windows.Input;
+
+namespace WPFCommandFactory
+{
+    internal class Command : ICommand
+    {
+        private readonly Action<object> _execute;
+        private readonly Func<object, bool> _canExecute;
+
+        public event EventHandler CanExecuteChanged
+        {
+            add { CommandManager.RequerySuggested += value; }
+            remove { CommandManager.RequerySuggested -= value; }
+        }
+
+        public Command(Action<object> execute, Func<object, bool> canExecute = null)
+        {
+            _execute = execute;
+            _canExecute = canExecute;
+        }
+
+        public bool CanExecute(object parameter)
+        {
+            return _canExecute == null || _canExecute(parameter);
+        }
+
+        public void Execute(object parameter)
+        {
+            _execute(parameter);
```

Comment:
```suggestion
           if(_canExecute(parameter))
                _execute(parameter);
```
Перед выполнением лучше проверить ещё раз